repo-scripts/repo-toolbox/src/cli/actions/CollectJsonSchemasAction.ts

```diff
@@ -0,0 +1,138 @@
+): AsyncIterable<IFolderItemToCopy> {
+  let folderItems: FolderItem[];
+  try {
+    folderItems = await FileSystem.readFolderItemsAsync(folderAbsolutePath);
+  } catch (e) {
+    if (!FileSystem.isNotExistError(e)) {
+      throw e;
+    } else {
+      return;
+    }
+  }
+
+  for (const entry of folderItems) {
+    const entryRelativePath: string = `${folderRelativePath}/${entry.name}`;
+    const entryAbsolutePath: string = `${folderAbsolutePath}/${entry.name}`;
+    if (entry.isDirectory()) {
+      yield* _getFolderItemsRecursiveAsync(entryAbsolutePath, entryRelativePath);
+    } else {
+      const content: string = await FileSystem.readFileAsync(entryAbsolutePath);
+      yield {
+        absolutePath: entryAbsolutePath,
+        relativePath: entryRelativePath,
+        content
+      };
+    }
+  }
+}
+
+export class CollectJsonSchemasAction extends CommandLineAction {
+  private readonly _outputPathParameter: IRequiredCommandLineStringParameter;
+
+  private readonly _terminal: ITerminal;
+
+  public constructor(terminal: ITerminal) {
+    super({
+      actionName: 'collect-json-schemas',
+      summary: 'Generates JSON schema files based on rush.json inventory',
+      documentation: "Use this to update the repo's JSON schema files"
+    });
+
+    this._terminal = terminal;
+
+    this._outputPathParameter = this.defineStringParameter({
+      parameterLongName: '--output-path',
+      description: 'Path to the output directory for the generated JSON schema files.',
+      argumentName: 'PATH',
+      required: true
+    });
+  }
+
+  protected override async onExecuteAsync(): Promise<void> {
+    const terminal: ITerminal = this._terminal;
+    const rushConfiguration: RushConfiguration = RushConfiguration.loadFromDefaultLocation();
+
+    const outputPath: string = path.resolve(this._outputPathParameter.value);
+
+    const contentByAbsolutePathByRelativePath: Map<string, Map<string, string[]>> = new Map();
+
+    await Async.forEachAsync(
+      rushConfiguration.projects,
+      async ({ projectFolder }: RushConfigurationProject) => {
+        const schemaFiles: AsyncIterable<IFolderItemToCopy> = _getFolderItemsRecursiveAsync(
+          `${projectFolder}/temp/json-schemas`,
+          ''
+        );
+        await Async.forEachAsync(schemaFiles, async ({ absolutePath, relativePath, content }) => {
+          let contentByAbsolutePath: Map<string, string[]> | undefined =
+            contentByAbsolutePathByRelativePath.get(relativePath);
+          if (!contentByAbsolutePath) {
+            contentByAbsolutePath = new Map();
+            contentByAbsolutePathByRelativePath.set(relativePath, contentByAbsolutePath);
+          }
+
+          let absolutePaths: string[] | undefined = contentByAbsolutePath.get(content);
+          if (!absolutePaths) {
+            absolutePaths = [];
+            contentByAbsolutePath.set(content, absolutePaths);
+          }
+
+          absolutePaths.push(absolutePath);
+        });
```

Comment:
Set concurrency to 1 since the actual logic isn't asynchronous?

Reply:
```suggestion
        await Async.forEachAsync(schemaFiles, async ({ absolutePath, relativePath, content }) => {
          let contentByAbsolutePath: Map<string, string[]> | undefined =
            contentByAbsolutePathByRelativePath.get(relativePath);
          if (!contentByAbsolutePath) {
            contentByAbsolutePath = new Map();
            contentByAbsolutePathByRelativePath.set(relativePath, contentByAbsolutePath);
          }

          let absolutePaths: string[] | undefined = contentByAbsolutePath.get(content);
          if (!absolutePaths) {
            absolutePaths = [];
            contentByAbsolutePath.set(content, absolutePaths);
          }

          absolutePaths.push(absolutePath);
        }, { concurrency: 5 });
```